File: backend/app/schemas/conversation.py
from datetime import datetime

from pydantic import BaseModel


class ConversationSummary(BaseModel):
    id: int
    mode: str
    title: str
    created_at: datetime
    updated_at: datetime


class MessageResponse(BaseModel):
    id: int
    role: str
    content: str
    created_at: datetime


class ConversationListResponse(BaseModel):
    conversations: list[ConversationSummary]


class MessageListResponse(BaseModel):
    conversation_id: int
    messages: list[MessageResponse]

File: frontend/src/types/conversation.ts
export type ConversationSummary = {
  id: number;
  mode: string;
  title: string;
  created_at: string;
  updated_at: string;
};

export type MessageDetail = {
  id: number;
  role: 'user' | 'assistant';
  content: string;
  created_at: string;
};

export type ConversationListResponse = { conversations: ConversationSummary[] };

export type MessageListResponse = { conversation_id: number; messages: MessageDetail[] };
